x-pack/test/functional/page_objects/security_page.ts

```diff
@@ -295,18 +295,37 @@ export class SecurityPageObject extends FtrService {
     return user as AuthenticatedUser;
   }
 
-  async forceLogout() {
+  async forceLogout(
+    { waitForLoginPage }: { waitForLoginPage: boolean } = { waitForLoginPage: true }
+  ) {
     this.log.debug('SecurityPage.forceLogout');
     if (await this.find.existsByDisplayedByCssSelector('.login-form', 100)) {
       this.log.debug('Already on the login page, not forcing anything');
       return;
     }
 
-    this.log.debug('Redirecting to /logout to force the logout');
+    this.log.debug(`Redirecting to ${this.deployment.getHostPort()}/logout to force the logout`);
     const url = this.deployment.getHostPort() + '/logout';
     await this.browser.get(url);
-    this.log.debug('Waiting on the login form to appear');
-    await this.waitForLoginPage();
+
+    // After logging out, the user can be redirected to various locations depending on the context. By default, we
+    // expect the user to be redirected to the login page. However, if the login page is not available for some reason,
+    // we should simply wait until the user is redirected *elsewhere*.
+    if (waitForLoginPage) {
+      this.log.debug('Waiting on the login form to appear');
+      await this.waitForLoginPage();
+    } else {
+      this.log.debug('Waiting for logout to complete');
+      await this.retry.waitFor('Waiting for logout to complete', async () => {
+        // There are cases when browser/Kibana would like users to confirm that they want to navigate away from the
+        // current page and lose the state (e.g. unsaved changes) via native alert dialog.
+        const alert = await this.browser.getAlert();
+        if (alert?.accept) {
+          await alert.accept();
+        }
+        return !(await this.browser.getCurrentUrl()).includes('/logout');
```

Comment:
I'm wondering if leaving the method here might introduce flakiness. Tests might assume that when `forceLogout` returns, they can actually move on interacting with Kibana (it worked like this in the past when we waited for the login page). I think the fact that the browser URL changed doesn't necessarily mean the `elsewhere` page is actually done loading / in some kind of ready state for further interaction.

Reply:
> Tests might assume that when forceLogout returns, they can actually move on interacting with Kibana

Well, this method doesn't imply any particular end state as it's different from environment to environment. Depending on the configuration and environment it can be `/login`, `/logged_out`, `/` or external Elastic Cloud projects page (the case for Serverless tests). So after calling this method consumers are supposed to navigate to the page they need explicitly since they cannot and shouldn't interact with Kibana until they log in again.

> it worked like this in the past when we waited for the login page

Yeah, but it's internal implementation detail of the entire logout flow that only Security page can rely on internally, other consumers shouldn't.

> I think the fact that the browser URL changed doesn't necessarily mean the elsewhere page is actually done loading / in some kind of ready state for further interaction.

If user is redirected from `/logout` to any other URL it always means that the cookie has been invalidated and user is no longer logged in. That's the only goal of this method. Do you have some specific case in mind when it'd be problematic?

Reply:
Having said that, I'm happy to adopt any other approach if you have suggestions, I just couldn't think of anything else that would work for serverless when user is redirected to the Elastic Cloud projects list after logout.